### PR TITLE
Player Blindness status

### PR DESCRIPTION
**Pull request by chucksellick**
Adds a new Blind player enchantment which can be inflicted with a wand of light, instead of confusion. Other existing sources of monster blindness such as darts of atropa will also now use the new status when used on the player.

Under this effect, your accuracy is greatly reduced, and increasingly worse the further away the target is. Mechanically your "to hit" roll is reduced by 10% for every tile distance from the player, starting at 10% on your own tile. 80% malus at max range for a normal species, 90% for Barachi. See individual commits for further detail.

Some additional sources of Blindness are also added:
  * It's in the possible chaos brand effects (in the "rare" list) - for weapons and clouds
  * Klown pies now have a chance to get glitter in your eyes (for now there's an artificially high chance for this particular pie, but will be lowered after some testing and feedback)
  * Zin wrath can blind you with silver when reciting at you
  * Gozag wrath can dazzle you at the sight of coin (only when dropped on monster death, it's more interesting if it happens during combat)

Not done and can be done in separate PRs:
  * Dazzling Flash is still a player-only spell so ghosts will not be blinding you yet
  * Perhaps the visual effect can be improved (with a tiles overlay) but I think it's ok as a placeholder
  * Vhi's Electric Charge has a slight issue as it reports the to-hit from your current position but actually it'll be more accurate as you'll be close

### CODE REVIEW

**Comment by pooka109** [comment]
Since you've put dockerizing webtiles into its own PR by now, shouldn't it be removed from this one?

File: crawl-ref/source/duration-data.h [hunk 1]
```diff
@@ -636,6 +636,10 @@ static const duration_def duration_data[] =
     { DUR_BEOGH_CAN_RECRUIT, LIGHTBLUE, "Recruit", "", "can recruit",
       "You may recruit a defeated apostle into your service", D_EXPIRES,
        {{ "", end_beogh_recruit_window}}},
+    { DUR_BLIND, RED, "Blind", "blind", "blindness",
+      "Your combat accuracy is reduced more the further away your target is,"
+      " and range is reduced for many spells.", D_EXPIRES, {{ "",
```
[the remark this answers is not in the document]
Comment (chucksellick) [reply crawl-ref/source/duration-data.h]: I did this (although I wasn't totally sure, as blindness is more a physical effect like poisoning, and comes from at least one poison source anyway, I thought maybe it should only be curable by potion...)

**Comment by chucksellick** [comment]
> Since you've put dockerizing webtiles into its own PR by now, shouldn't it be removed from this one?

Ah that was in by mistake because i was using it for testing

**Comment by DracoOmega** [comment]
Merged into trunk (with tweaks and fixes). Thanks for the work!

**Pull request closed without merging**